Feature/restaurant page/dish

talorlangnas-moveo opened the pull request.

talorlangnas-moveo committed:
adding dish images and creating dish interface

talorlangnas-moveo committed:
Merge branch 'feature/filters' into feature/restaurant_page/dish

talorlangnas-moveo committed:
Update Card and RestaurantsHome components for improved styling and functionality

- Added className prop to Card component for better customization.
- Enhanced rendering of restaurant cards in RestaurantsHome to utilize the new className.
- Updated styles in card.module.scss and restaurantsHome.module.scss for improved layout and responsiveness.
- Introduced slug property in restaurant data for dynamic routing.
- Fixed filter option ID formatting in filterFunctions.ts for consistency.

Still need to add a line next to the price and fix the card's height

talorlangnas-moveo committed:
Refactor Card and RestaurantHome components for improved styling and layout

- Updated card.module.scss to enhance padding, alignment, and responsiveness of card elements.

- Adjusted restaurantHomePage.module.scss for better layout and alignment on larger screens.

- Modified card.tsx to streamline the structure of price display elements.

talorlangnas-moveo committed:
Refactor restaurant components and remove unused files

- Replaced RestaurantsHome component with RestaurantsDisplay for improved structure and styling.

- Removed unused claroDishData and restaurant slug properties from the restaurant data interface for clarity.

- Updated fetch functions to align with the new component structure and removed unnecessary imports.

talorlangnas-moveo committed:
fix filterFunctions file name

taluliel18 commented on epicure-app/src/components/restaurantsDisplay/restaurantsDisplay.tsx:
```diff
@@ -69,11 +72,16 @@ export default function RestaurantsHome({
-            {...restaurant}
-            className={styles.restaurantCard}
-          />
+            href={`/restaurants/${restaurant.id}?id=${restaurant.id}`}
```
You’re already passing the ID in the route — there’s no need to also use ?id=${restaurant.id}

talorlangnas-moveo replied:
done

talorlangnas-moveo committed:
Refactor RestaurantPage and RestaurantsDisplay components for improved parameter handling

- Updated RestaurantPage to use 'params' instead of 'searchParams' for fetching restaurant and dish data.

- Simplified the URL structure in RestaurantsDisplay by removing redundant query parameters in the link to restaurant pages.
